### PR TITLE
feat: support eslint 8

### PR DESCRIPTION
**Pull request by ShadyZOZ**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

Updated dependencies to support eslint 8.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,9 +47,9 @@
     "contributor": "git-contributor"
   },
   "engines": {
-    "node": ">=10.0.0"
+    "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
```
Comment (fengmk2) [inline package.json:50]: 发 major ，可以的。

**Review by fengmk2 — APPROVED** [review]
+1

**Review by whxaxes — APPROVED** [review]

**Review by hyj1991 — APPROVED** [review]
+1

**Comment by fengmk2** [comment]
10.0.0 released

File: package.json [hunk 2]
```diff
@@ -12,24 +12,26 @@
     "lib"
   ],
   "dependencies": {
-    "@typescript-eslint/eslint-plugin": "^3.0.0",
-    "@typescript-eslint/parser": "^3.0.0",
-    "babel-eslint": "^8.2.6",
+    "@babel/core": "^7.16.0",
```
Comment (fengmk2) [inline package.json:15]: 为什么会加上 babel 依赖？ @ShadyZOZ 
@atian25 还记得么？

**Pull request merged**